packages/playground/src/components/ssh_keys/SshDataDialog.vue

```diff
@@ -74,11 +82,44 @@ export default defineComponent({
       required: true,
     },
   },
-  setup() {
+  setup(props, ctx) {
+    const currentKey = ref<SSHKeyData>(props.selectedKey);
+    const loading = ref<boolean>(false);
+
+    watch(
+      () => props.open,
+      newValue => {
+        if (newValue) {
+          currentKey.value = { ...props.selectedKey };
+        }
+      },
+    );
+
     const notNeededFields = ["id", "activating", "deleting", "isActive", "createdAt"];
+    const sshKeysManagement = new SSHKeysManagement();
+    const updateKey = () => {
+      loading.value = true;
+      ctx.emit("update", currentKey);
+      ctx.emit("close");
```

Comment:
I think we need to close the dialog from the parent component after calling the chain, this will provide a better experience, e.g. we open the modal while we updating the chain and just close it after the successful update